src/app/server/Server.cpp

```diff
@@ -127,11 +129,13 @@ CHIP_ERROR Server::Init(const ServerInitParams & initParams)
     SuccessOrExit(mAttributePersister.Init(mDeviceStorage));
     SetAttributePersistenceProvider(&mAttributePersister);
 
-    InitDataModelHandler(&mExchangeMgr);
-
     err = mFabrics.Init(mDeviceStorage);
     SuccessOrExit(err);
 
+    SuccessOrExit(err = mAccessControl.Init(initParams.accessDelegate, sDeviceTypeResolver));
+    Access::SetAccessControl(mAccessControl);
+    SuccessOrExit(err = mAclStorage.Init(*mDeviceStorage, mFabrics));
```

Comment:
Let's make AclStorage an interface and inject it in initParams, so it can be overridden if needed. Can be done as a follow-up

Reply:
Mentioned in #17912.